src/components/flat-table/flat-table-expandable.stories.mdx

```diff
@@ -0,0 +1,972 @@
+import { Meta, Props, Preview, Story } from "@storybook/addon-docs/blocks";
+import { useState, useRef } from "react";
+import StyledSystemProps from "../../../.storybook/utils/styled-system-props";
+import {
+  FlatTable,
+  FlatTableHead,
+  FlatTableBody,
+  FlatTableRow,
+  FlatTableHeader,
+  FlatTableRowHeader,
+  FlatTableCell,
+  FlatTableCheckbox,
+  Sort,
+} from ".";
+import BatchSelection from "../batch-selection";
+import IconButton from "../icon-button";
+import Icon from "../icon";
+import Textbox from "../../__experimental__/components/textbox";
+import Pager from "../pager";
+import {
+  ActionPopover,
+  ActionPopoverItem,
+  ActionPopoverMenu,
+} from "../action-popover";
+import { SidebarContext } from "../drawer";
+import Box from "../box";
+
+<Meta
+  title="Design System/Flat Table/Expandable"
+  parameters={{ info: { disable: true } }}
+/>
+
+## Flat Table expandable options
+
+### Default
+
+Flat table rows can be made expandable, with any number of sub rows.
+
+<Preview>
+  <Story name="default">
+    {() => {
+      const SubRows = [
```

Comment:
These `SubRows` arrays need keys or there will be console errors in storybook when the stories are viewed

Reply:
They don't actually, and there's no console errors

Reply:
When you use `React.Children.map` it adds an implicit key @edleeks87

Reply:
Fair point, I had a console warning but I'm thinking it's due to the npm woes I'm having at the minute and related to something else in one of the other flat table stories